src/KeyFrameDatabase.cc

```diff
@@ -709,20 +709,19 @@ void KeyFrameDatabase::DetectNBestCandidates(KeyFrame *pKF, vector<KeyFrame*> &v
     while(i < lAccScoreAndMatch.size() && (vpLoopCand.size() < nNumCandidates || vpMergeCand.size() < nNumCandidates))
     {
         KeyFrame* pKFi = it->second;
-        if(pKFi->isBad())
-            continue;
-
-        if(!spAlreadyAddedKF.count(pKFi))
-        {
-            if(pKF->GetMap() == pKFi->GetMap() && vpLoopCand.size() < nNumCandidates)
-            {
-                vpLoopCand.push_back(pKFi);
-            }
-            else if(pKF->GetMap() != pKFi->GetMap() && vpMergeCand.size() < nNumCandidates && !pKFi->GetMap()->IsBad())
+        if(pKFi->isBad()){
```

Comment:
If you want to execute the code inside when the KF is not bad then wouldn't this be:
```suggestion
        if(!pKFi->isBad()){
```
or rather simply keeping the code as it was before, but adding an increment to the iterator:
```suggestion
        if(pKFi->isBad(){
            it++;
            continue;
        }
```
?

Reply:
hhh, yeah, I made a new bug here ...